History/HistoryManager/src/main/java/eu/neclab/ngsildbroker/historymanager/repository/HistoryDAO.java

```diff
@@ -41,7 +42,48 @@ public void entityExists(String entityId, String tenantId) throws ResponseExcept
 
 		}
 	}
-	
+
+	public void attributeExists(String entityId, String tenantId, String resolvedAttrId, String instanceId)
+			throws ResponseException {
+		List<String> result;
+		List<String> instanceattrresult;
+		if (tenantId == AppConstants.INTERNAL_NULL_KEY) {
+			result = getJDBCTemplate(null).queryForList(
+					"select attributeid from temporalentityattrinstance WHERE temporalentity_id='" + entityId + "'",
+					String.class);
+			if (!result.contains(resolvedAttrId)) {
+				throw new ResponseException(ErrorType.NotFound, resolvedAttrId + " not found");
+			} else {
+				if (instanceId != null) {
+					instanceattrresult = getJDBCTemplate(null)
+							.queryForList("select attributeid from temporalentityattrinstance WHERE temporalentity_id='"
+									+ entityId + "' and instanceid='" + instanceId + "'", String.class);
+					if (!instanceattrresult.contains(resolvedAttrId)) {
+						throw new ResponseException(ErrorType.NotFound, instanceId + " not found");
+					}
+				}
+			}
+		} else {
+			if (instanceId == null) {
+				result = getJDBCTemplate(tenantId).queryForList(
+						"select attributeid from temporalentityattrinstance WHERE temporalentity_id='" + entityId + "'",
+						String.class);
+				if (!result.contains(resolvedAttrId)) {
+					throw new ResponseException(ErrorType.NotFound, resolvedAttrId + " not found");
+				}
+			} else {
+				if (instanceId != null) {
+					instanceattrresult = getJDBCTemplate(null)
+							.queryForList("select attributeid from temporalentityattrinstance WHERE temporalentity_id='"
+									+ entityId + "' and instanceid='" + instanceId + "'", String.class);
+					if (!instanceattrresult.contains(resolvedAttrId)) {
+						throw new ResponseException(ErrorType.NotFound, instanceId + " not found");
+					}
+				}
+			}
+		}
+	}
```

Comment:
So fix that issue like as "instance id is not exist" or "attribute id is not" whatever so you can take HashMap<String,List<String>> so for that logic no need to add if else condition and also no need to take multiple query.

Reply:
Done

Reply:
please have a look to solve this within the storageDAO similar to the already exists where you check against the number of lines changed on the actual sql request. in the quarkus solution we should be able to check this in onfailure.